### ISSUE TITLE
fix: @Inject conflicts with reflect-metadata

### ISSUE THREAD

**Issue by synchrone** [typestack/typedi#102]
A combination of name-based @Inject(), an inherited constructor and a class that is defined in a module that happens to import 'reflect-metadata' results in that dependency not being injected.
```typescript
import 'reflect-metadata'
import {Container, Inject, Service} from "typedi";

@Service()
export class Bla {}

const testconfig = {}
type ConfigType = typeof testconfig

@Service()
export class Base {
  public constructor(
    @Inject('config')
    protected cfg: ConfigType, //This is not injected
    protected bla: Bla // if this is coming from a module w/ import 'reflect-metadata'
  ) {
    if(!cfg){
      throw Error('expected typedi to inject cfg')
    }
    console.log('cfg injected')
  }
}

@Service()
export class Child extends Base {}


if (require.main === module) {
  Container.set('config', testconfig)
  Container.get(Base) //works
  Container.get(Child) //inherited constructor does not receive a named dependency
}
```

Reproduction repo in https://gist.github.com/synchrone/0fe69a1680fd686e5438b13a4c8d2134

**Comment by NoNameProvided** [comment]
Confirmed, however, it's not related to reflect-metadata.

**Comment by NoNameProvided** [comment]
Fixed in 973a920b8f0d1b4c0f0014410fbdce6191324479. Will be included in the next release.

**Comment by NoNameProvided** [comment]
This has been released as `0.9.0`.
